src/tests/e2etest/ZtsiTests.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Azure.Devices;
+using System.Text.Json;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace e2etesting
+{
+    [TestFixture]
+    public class ZtsiTests : E2eTest
+    {   
+        string ComponentName = "Ztsi";
+        public partial class Ztsi
+        {
+            public int Enabled { get; set; }
+            public string ServiceUrl { get; set; }
+        }
+        public partial class DesiredZtsi
+        {
+            public bool DesiredEnabled { get; set; }
+            public string DesiredServiceUrl { get; set; }
+        }
+
+        public partial class ResponseCode
+        {
+            public int ac { get; set; }
+        }
+        private int m_twinTimeoutSeconds;
+
+        [SetUp]
+        public void TestSetUp()
+        {
+            m_twinTimeoutSeconds = base.twinTimeoutSeconds;
+            base.twinTimeoutSeconds = base.twinTimeoutSeconds * 2;
+        }
+
+        [TearDown]
+        public void TestTearDown()
+        {
+            base.twinTimeoutSeconds = m_twinTimeoutSeconds;
+        }
+
+        [Test]      
+        public void ZtsiTest_Get()
+        {
+            if ((GetTwin().ConnectionState == DeviceConnectionState.Disconnected) || (GetTwin().Status == DeviceStatus.Disabled))
```

Comment:
Recommend using `GetNewTwin` to ensure most up-to-date twin in the first argument of the conditional